components/insights/related-posts.tsx

```diff
@@ -0,0 +1,80 @@
+import { InsightPost, insights } from '@/data/insights'
+import Image from 'next/image'
+import Link from 'next/link'
+
+interface RelatedPostsProps {
+  currentPostId: string
+  category: string
+}
+
+export default function RelatedPosts({
+  currentPostId,
+  category,
+}: RelatedPostsProps) {
+  // Get related posts (same category, excluding current post)
+  const relatedPosts = insights
+    .filter((post) => post.category === category && post.id !== currentPostId)
+    .slice(0, 3) // Limit to 3 related posts
+
+  if (relatedPosts.length === 0) {
+    return null
+  }
+
+  return (
+    <section className='mt-16 border-t border-gray-200 pt-12'>
+      <h2 className='mb-8 text-2xl font-bold text-gray-900'>
+        Related Insights
+      </h2>
+
+      <div className='grid grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3'>
+        {relatedPosts.map((post) => (
+          <RelatedPostCard key={post.id} post={post} />
+        ))}
+      </div>
+    </section>
+  )
+}
+
+interface RelatedPostCardProps {
+  post: InsightPost
+}
+
+function RelatedPostCard({ post }: RelatedPostCardProps) {
```

Comment:
please break this out into it's own file. I can think of at least one other place on the site where we might want to use relatedPostCard so I would not recommend to localise it in related-posts